host/4.0/buster/amd64/dotnet/dotnet-inproc/dotnet.Dockerfile

```diff
@@ -0,0 +1,34 @@
+# Build the runtime from source
+ARG HOST_VERSION=3.0.15584
+FROM mcr.microsoft.com/dotnet/sdk:6.0 AS runtime-image
+ARG HOST_VERSION
+
+# Build requires 3.1 SDK
+COPY --from=mcr.microsoft.com/dotnet/core/sdk:3.1 /usr/share/dotnet /usr/share/dotnet
```

Comment:
Since you need a different version of dotnet for building and runtime, you can try something similar to here - https://github.com/Azure/azure-functions-docker/blob/dev/host/2.0/stretch/amd64/dotnet/dotnet-appservice.Dockerfile#L33
This way you can use the dotnet 3.1 image for and then add another layer for dotnet 6.0 for runtime.

Reply:
Well, it's a little trickier. We need both .NET 6 and .NET 3.1 SDKs installed in order to build -- this should be temporary as we fix up our build targets in the near future. And then we'll run on .NET 6 only.

This step is copying the sdk bits from the .NET 3.1 so that both SDKs are present during the build